training-1/src/main.rs

```diff
@@ -126,8 +131,11 @@ fn main() {
         }
 
         if !level_complete {
-            if let Some(cmd) = vm.console.try_process_command() {
-                vm.execute_command(cmd);
+            if let Some(cmd) = vm.console.get_next_command() {
```

Comment:
I still really wish this could be moved inside the VM. I think you were saying that you were thinking of wrapping it all in another third type?